src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -116,19 +112,28 @@ export function loadOperation(
             typespecParameters.body.type
         );
         if (effectiveBodyType.kind === "Model") {
-            if (effectiveBodyType.name !== "") {
-                parameters.push(
-                    loadBodyParameter(sdkContext, effectiveBodyType)
-                );
-            } else {
-                effectiveBodyType.name = `${capitalize(op.name)}Request`;
-                let bodyParameter = loadBodyParameter(
-                    sdkContext,
-                    effectiveBodyType
-                );
+            let bodyParameter = loadBodyParameter(
+                sdkContext,
+                effectiveBodyType
+            );
+            if (effectiveBodyType.name == "") {
```

Comment:
use  '==='  here

Reply:
We're excluding zero value here, so I tend to use `==`.

Reply:
What is zero value?

Reply:
> What is zero value?

undefined, null, 0, false, etc. I think if we compare with a meaningful value, like `"abc"`, it should use `===`.

Reply:
if you intentional want to check "if the name is empty", we should use `if (!effectiveBodyType.name)`
And since we are using typescript, I believe it is over thinking that we could get `0` or `false` or others. We should always use `===`.

Reply:
rolled back to `===`